aws-s3-transfer-manager/src/operation/download/handle.rs

```diff
@@ -53,29 +53,26 @@ impl DownloadHandle {
         Ok(meta)
     }
 
-    /// Object content
-    pub fn body(&self) -> &Body {
-        &self.body
+    /// Object content and metadata
+    pub fn output(&self) -> &Output {
+        &self.output
     }
 
-    /// Mutable reference to the body
-    pub fn body_mut(&mut self) -> &mut Body {
-        &mut self.body
+    /// Mutable reference to the output
+    pub fn ouput_mut(&mut self) -> &mut Output {
+        &mut self.output
     }
 
-    /// Consume the handle and wait for download transfer to complete
-    #[tracing::instrument(skip_all, level = "debug", name = "join-download")]
-    pub async fn join(mut self) -> Result<(), crate::error::Error> {
-        self.body.close();
-
-        self.discovery.await??;
+    /// Abort the download and cancel any in-progress work.
+    pub async fn abort(mut self) {
+        self.output.close();
+        self.discovery.abort();
+        let _ = self.discovery.await;
```

Comment:
Oh, it's because we're dropping an error using `let _ = self.discovery.await`. Instead, I was referring to
```
if let Err(e) = self.discovery.await {
  // debug log `e` for the record
}
```

> why would the discovery task panic?

I don't know either (according to [the doc](https://docs.rs/tokio/latest/tokio/task/struct.JoinHandle.html#examples) it's due to panics), but because of that, I was thinking whether at least it records that fact as a debug log. Maybe my suggestion is instead of ignoring the result by `let _ =` we should use either `expect` or debug log that error.

Reply:
The result in most cases would be JoinedError as stated in the docs:
```
Abort the task associated with the handle.

Awaiting a cancelled task might complete as usual if the task was already completed at the time it was cancelled, but most likely it will fail with a [cancelled](https://docs.rs/tokio/latest/tokio/task/struct.JoinError.html#method.is_cancelled) JoinError.
```

So I think logging the error here will not be useful. Even if it's not this error, we don't care about errors if we are aborting the download.